presto-main/src/main/java/com/facebook/presto/operator/aggregation/BigWriteableBlock.java

```diff
@@ -0,0 +1,307 @@
+    {
+        int block = decodeSliceIndex(addresses.get(position));
+        int blockPosition = decodePosition(addresses.get(position));
+        return blocks.get(block).bytesCompare(blockPosition, offset, length, otherSlice, otherOffset, otherLength);
+    }
+
+    @Override
+    public void writeBytesTo(int position, int offset, int length, BlockBuilder blockBuilder)
+    {
+        int block = decodeSliceIndex(addresses.get(position));
+        int blockPosition = decodePosition(addresses.get(position));
+        blocks.get(block).writeBytesTo(blockPosition, offset, length, blockBuilder);
+    }
+
+    @Override
+    public boolean equals(int position, int offset, Block otherBlock, int otherPosition, int otherOffset, int length)
+    {
+        int block = decodeSliceIndex(addresses.get(position));
+        int blockPosition = decodePosition(addresses.get(position));
+        return blocks.get(block).equals(blockPosition, offset, otherBlock, otherPosition, otherOffset, length);
+    }
+
+    @Override
+    public int hash(int position, int offset, int length)
+    {
+        int block = decodeSliceIndex(addresses.get(position));
+        int blockPosition = decodePosition(addresses.get(position));
+        return blocks.get(block).hash(blockPosition, offset, length);
+    }
+
+    @Override
+    public int compareTo(int leftPosition, int leftOffset, int leftLength, Block rightBlock, int rightPosition, int rightOffset, int rightLength)
+    {
+        int block = decodeSliceIndex(addresses.get(leftPosition));
+        int blockPosition = decodePosition(addresses.get(leftPosition));
+        return blocks.get(block).compareTo(blockPosition, leftOffset, leftLength, rightBlock, rightPosition, rightOffset, rightLength);
+    }
+
+    @Override
+    public Block getSingleValueBlock(int position)
+    {
+        int block = decodeSliceIndex(addresses.get(position));
+        int blockPosition = decodePosition(addresses.get(position));
+        return blocks.get(block).getSingleValueBlock(blockPosition);
+    }
+
+    @Override
+    public int getPositionCount()
+    {
+        return nextPosition;
+    }
+
+    @Override
+    public int getSizeInBytes()
+    {
+        return completedBlockSize + currentBuilder.getSizeInBytes();
+    }
+
+    @Override
+    public BlockEncoding getEncoding()
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Block getRegion(int positionOffset, int length)
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public boolean isNull(int position)
+    {
+        int block = decodeSliceIndex(addresses.get(position));
+        int blockPosition = decodePosition(addresses.get(position));
+        return blocks.get(block).isNull(blockPosition);
+    }
+
+    @Override
+    public void assureLoaded()
+    {
+        currentBuilder.assureLoaded();
```

Comment:
this isn't needed.  assure loaded is for lazy blocks and this is not a lazy block